### PR TITLE
Add max decimal place option to recorder.

### PR DESCRIPTION
**Pull request by 0mdc**
## Motivation and Context

This changes the gfx-replay `Recorder` to make the maximum float decimal places configurable.

This is an easy way to increase performance of JSON-based systems, as well as making the output text more compact.

**Example usage:**
```
sim.gfx_replay_manager.set_max_decimal_places(3)
[...]
sim.gfx_replay_manager.save_keyframe()
keyframe = sim.gfx_replay_manager.write_saved_keyframes_to_string()
```

**Example data (update frames):**
| Default | 3 Decimal Places |
|---|---|
| 4223 bytes| 3656 bytes |
| jsonbefore | jsonafter |

## How Has This Been Tested

Added a unit test, and tested locally on the HITL tool.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/tests/GfxReplayTest.cpp [hunk 1]
```diff
@@ -722,6 +724,54 @@ void GfxReplayTest::testSkinningIntegration() {
...
+    auto keyframe = recorder->debugGetSavedKeyframes().front();
+    keyframe3Decimals =
+        recorder->writeIncrementalSavedKeyframesToStringArray().front();
+    keyframe5Decimals =
+        esp::gfx::replay::Recorder::keyframeToString(keyframe, 5);
+  }
+
+  // read the playback strings
+  {
+    CORRADE_VERIFY(keyframe3Decimals.find("0.555") != std::string::npos);
```
Comment (mosra) [inline src/tests/GfxReplayTest.cpp:768]: Just FYI, there's [TestSuite::Compare::StringContains](https://doc.magnum.graphics/corrade/classCorrade_1_1TestSuite_1_1Compare_1_1StringContains.html) and also [TestSuite::Compare::StringNotContains](https://doc.magnum.graphics/corrade/classCorrade_1_1TestSuite_1_1Compare_1_1StringNotContains.html) for better failure diagnostics.

Comment (0mdc) [reply src/tests/GfxReplayTest.cpp:768]: Much cleaner! Thank you!

File: src/esp/gfx/replay/Recorder.h [hunk 2]
```diff
@@ -150,17 +152,31 @@ class Recorder {
...
+   * @brief Get the precision of the floating points serialized by this
+   * recorder.
+   */
+  int getMaxDecimalPlaces() const;
+
   /**
    * @brief returns JSONized version of given keyframe.
    */
-  static std::string keyframeToString(const Keyframe& keyframe);
+  static std::string keyframeToString(
```
Comment (eundersander) [inline src/esp/gfx/replay/Recorder.h:170]: This looks really error-prone. I can imagine calling `recorder.setDecimalPlaces(3)`, then calling `Recorder::keyframeToString(keyframe)`, and not understanding what I did wrong.

How about we switch `keyframeToString` to be non-static? Alternately, at least make `maxDecimalPlaces` a required arg.

**Review by eundersander — APPROVED** [review]
Thanks for adding this, including the unit test. I have one request below, otherwise LGTM!

File: src/esp/gfx/replay/Recorder.h (hunk 2 above)
Comment (0mdc) [reply src/esp/gfx/replay/Recorder.h]: Agreed. This is much better.

**Pull request merged**